src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -457,27 +457,54 @@ func (r *StatefulSetResource) obj() (k8sclient.Object, error) {
 	return ss, nil
 }
 
-func overprovisioned(developerMode bool, limits corev1.ResourceList) []string {
-	memory, exist := limits["memory"]
-	if !exist {
-		memory = resource.MustParse("2Gi")
-	}
+func prepareAdditionalArguments(
+	developerMode bool, originalRequests corev1.ResourceList,
+) []string {
+	requests := originalRequests.DeepCopy()
+
+	requests.Cpu().RoundUp(0)
+	requestedCores := requests.Cpu().Value()
+	requestedMemory := requests.Memory().Value()
```

Comment:
As this PR needs some fixes to pass CI, could you subtract from `requestedMemory` around 10% of the `requestedMemory`?

Reply:
I think you're asking to subtract 10% so the `--memory` doesn't end up taking the whole machine. Instead, the caller can ensure the passed requested memory is 90% (or other percent they prefer) of the machine capacity.

Reply:
I'm asking for add another layer of cushion. Because in the container there are more processes than Redpanda under heavy lead Redpanda will be oom killed. If we set request and limit for the stateful to 10 Gi I would like to have Redpanda `memory` argument to be 10% than 10Gi. I'm ok with doing this in the next PR.

Reply:
> If we set request and limit for the stateful to 10 Gi

With this PR we're not setting a Limit anymore, only Request and `--memory`

> I would like to have Redpanda memory argument to be 10% than 10Gi.

1) It's hard to decide on the value of "10%", why not "15%". In that case we'd need to not hardcode it and expose a parameter. 2) Most importantly, we already have "requests", so the user can enter a request that is 90% (or similar) of their capacity. We already do this when using the operator.

Reply:
> 1. It's hard to decide on the value of "10%", why not "15%". In that case we'd need to not hardcode it and expose a parameter.

The values was calculated by me when the OOM Killed event was reported via dmesg. User should not provide this value, because this is Redpanda container problem. 

> 2. Most importantly, we already have "requests", so the user can enter a request that is 90% (or similar) of their capacity. We already do this when using the operator.

The requests on the cgroup level is different knob than Redpanda `--memory`.

Reply:
This PR passes the requested memory as the `--memory`